src/Storages/StorageDelta.cpp

```diff
@@ -0,0 +1,354 @@
+
+                    metadata.remove(path, timestamp);
+
+                    LOG_DEBUG(log, "Path {}", path);
+                    LOG_DEBUG(log, "Timestamp {}", timestamp);
+                }
+
+                // reset
+                opening = 0;
+                closing = 0;
+                json_str.clear();
+            }
+        }
+    }
+}
+
+std::vector<String> JsonMetadataGetter::getJsonLogFiles()
+{
+    std::vector<String> keys;
+
+    const auto & client = base_configuration.client;
+
+    Aws::S3::Model::ListObjectsV2Request request;
+    Aws::S3::Model::ListObjectsV2Outcome outcome;
+
+    bool is_finished{false};
+    const auto bucket{base_configuration.uri.bucket};
+
+    request.SetBucket(bucket);
+    request.SetPrefix(table_path + "_delta_log");
+
+    while (!is_finished)
+    {
+        outcome = client->ListObjectsV2(request);
+        if (!outcome.IsSuccess())
+            throw Exception(
+                ErrorCodes::S3_ERROR,
+                "Could not list objects in bucket {} with key {}, S3 exception: {}, message: {}",
+                quoteString(bucket),
+                quoteString(table_path),
+                backQuote(outcome.GetError().GetExceptionName()),
+                quoteString(outcome.GetError().GetMessage()));
+
+        const auto & result_batch = outcome.GetResult().GetContents();
+        for (const auto & obj : result_batch)
+        {
+            const auto & filename = obj.GetKey();
+
+            if (filename.substr(filename.size() - 5) == ".json")
+                keys.push_back(filename);
+        }
+
+        request.SetContinuationToken(outcome.GetResult().GetNextContinuationToken());
+        is_finished = !outcome.GetResult().GetIsTruncated();
+    }
+
+    return keys;
+}
+
+std::unique_ptr<ReadBuffer> JsonMetadataGetter::createS3ReadBuffer(const String & key)
+{
+
+    // TBD: add parallel downloads
+    return std::make_unique<ReadBufferFromS3>(
+        base_configuration.client,
+        base_configuration.uri.bucket,
+        key,
+        base_configuration.uri.version_id,
+        /* max single read retries */ 10,
+        ReadSettings{});
+}
+
+StorageDelta::StorageDelta(
+    const S3::URI & uri_,
+    const String & access_key_,
+    const String & secret_access_key_,
+    const StorageID & table_id_,
+    ColumnsDescription columns_,
+    const ConstraintsDescription & constraints_,
+    const String & comment,
+    ContextPtr context_)
+    : IStorage(table_id_)
+    , base_configuration({uri_, access_key_, secret_access_key_, {}, {}, {}})
+    , log(&Poco::Logger::get("StorageDeltaLake (" + table_id_.table_name + ")"))
+    , table_path(uri_.key)
+{
+    StorageInMemoryMetadata storage_metadata;
+    updateS3Configuration(context_, base_configuration);
+
+    JsonMetadataGetter getter{base_configuration, table_path, log};
+
+    auto keys = getter.getFiles();
+
+    for (const String & path : keys)
+    {
+        LOG_DEBUG(log, "{}", path);
```

Comment:
Need to remove this logging or add better logging message. And keys can be printed as `fmt::join(keys, ", ")`

Reply:
Removed unneeded logging